crates/proto/src/udp/udp_stream.rs

```diff
@@ -217,8 +217,11 @@ pub(crate) struct NextRandomUdpSocket<S> {
     bind_address: SocketAddr,
     closure: UdpCreator<S>,
     marker: PhantomData<S>,
+    state: Option<(usize, UdpSocketCreateFuture<S>)>,
```

Comment:
need some documentation here that the `usize` is the number of attempted socket creation attempts.

Reply:
alternatively, this could be a named type that is just an alias to `usize` to make it self-documenting.